releasenotes-builder/src/test/java/com/github/checkstyle/MainTest.java

```diff
@@ -19,102 +19,307 @@
 
 package com.github.checkstyle;
 
-import static org.kohsuke.github.GHIssueState.OPEN;
+import static org.kohsuke.github.GHIssueState.CLOSED;
 
+import org.junit.Assert;
 import org.junit.Test;
 
 import com.github.checkstyle.internal.AbstractReleaseNotesTestSupport;
 
 public class MainTest extends AbstractReleaseNotesTestSupport {
-
-    @Test
-    public void testValidateVersion() {
-        runMainAndAssertReturnCode(0,
-                "-releaseNumber", "10.0.1",
-                "-validateVersion"
-        );
-    }
-
     @Test
     public void testNoCommits() {
-        runMainAndAssertReturnCode(0,
-                "-localRepoPath", getTempFolder().getAbsolutePath(),
-                "-remoteRepoPath", "checkstyle/checkstyle",
-                "-startRef", "12345678",
-                "-releaseNumber", "10.0.1",
-                "-outputLocation", getTempFolder().getAbsolutePath(),
-                "-githubAuthToken", "TOKEN",
-                "-generateAll",
-                "-publishTwit",
-                "-twitterConsumerKey", "KEY",
-                "-twitterConsumerSecret", "SECRET",
-                "-twitterAccessToken", "TOKEN",
-                "-twitterAccessTokenSecret", "SECRET",
-                "-validateVersion"
+        runMainContentGenerationAndAssertReturnCode(0,
+            "-releaseNumber", "10.0.1",
+            "-generateAll",
+            "-validateVersion"
         );
+
+        Assert.assertEquals("expected error output", "", systemErr.getLog());
+        Assert.assertEquals("expected output", MSG_EXECUTION_SUCCEEDED, systemOut.getLog());
     }
 
     @Test
     public void testUnknownCommit() {
         addCommit("Hello World", "CheckstyleUser");
 
-        runMainAndAssertReturnCode(0,
-                "-localRepoPath", getTempFolder().getAbsolutePath(),
-                "-remoteRepoPath", "checkstyle/checkstyle",
-                "-startRef", "12345678",
-                "-releaseNumber", "10.0.1",
-                "-outputLocation", getTempFolder().getAbsolutePath(),
-                "-githubAuthToken", "TOKEN",
-                "-generateAll",
-                "-publishTwit",
-                "-twitterConsumerKey", "KEY",
-                "-twitterConsumerSecret", "SECRET",
-                "-twitterAccessToken", "TOKEN",
-                "-twitterAccessTokenSecret", "SECRET",
-                "-validateVersion"
+        runMainContentGenerationAndAssertReturnCode(0,
+            "-releaseNumber", "10.0.1.",
```

Comment:
How come this isn't an error? (extra period on end)

Reply:
`isPatch` is checking if the release number ends with a zero `0`. That is why.

Reply:
@romani Shouldn't `validateVersion` be checking this?

Reply:
validateVersion() calls isPatch()

Reply:
Let's move this to separate issue  , I will fix it later on

Reply:
https://github.com/checkstyle/contribution/issues/727

Reply:
Marking resolved